render-ws-java-client/src/main/java/org/janelia/render/client/parameter/MultiProjectParameters.java

```diff
@@ -127,6 +128,12 @@ public List<StackMFOVWithZValues> buildListOfStackMFOVWithAllZ(final String mult
         return stackMFOVWithZValuesList;
     }
 
+    public void setNamingGroup(final StackIdNamingGroup namingGroup) {
+        if ((namingGroup != null) && (this.stackIdWithZ != null)) {
```

Comment:
NamingGroups are optional parameters that should only be used if specified.  Null means unspecified.  This makes sense to me (more so than using a contrived pattern value), though I agree supporting optional parameters makes the code uglier.  Maybe I'm misunderstanding your point - so, let's discuss further as needed.

Reply:
I was thinking of the [null-object pattern](https://en.wikipedia.org/wiki/Null_object_pattern). An explicit null object (however that may look) could swallow a lot of the null-checks and optional logic. But happy to discuss this in person.

Reply:
I had a quick look and agree that the logic can probably not be trivially simplified by pulling logic into `StackIdNamingGroup`. However, I feel that just specifying `.*` as default pattern would get rid of a good deal of null-checks (since it automatically does what the null-patterns should represent).